File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariDFSSpaceRetrievalTask.java
package com.sequenceiq.cloudbreak.service.cluster.flow;

import java.util.Map;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import com.sequenceiq.cloudbreak.service.ClusterBasedStatusCheckerTask;

@Component
public class AmbariDFSSpaceRetrievalTask extends ClusterBasedStatusCheckerTask<AmbariClientPollerObject> {
    public static final int AMBARI_RETRYING_INTERVAL = 5000;
    public static final int AMBARI_RETRYING_COUNT = 3;

    private static final Logger LOGGER = LoggerFactory.getLogger(AmbariDFSSpaceRetrievalTask.class);
    private Map<String, Map<Long, Long>> dfsSpace;

    @Override
    public boolean checkStatus(AmbariClientPollerObject ambariClientPollerObject) {
        try {
            dfsSpace = ambariClientPollerObject.getAmbariClient().getDFSSpace();
            return true;
        } catch (Exception ex) {
            LOGGER.warn("Error during getting dfs space from ambari", ex);
            return false;
        }
    }

    @Override
    public void handleTimeout(AmbariClientPollerObject ambariClientPollerObject) {
    }

    @Override
    public String successMessage(AmbariClientPollerObject ambariClientPollerObject) {
        return "Dfs space successfully get from ambari.";
    }

    public Map<String, Map<Long, Long>> getDfsSpace() {
        return dfsSpace;
    }
}

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/flow/AmbariDFSSpaceRetrievalTask.kt
package com.sequenceiq.cloudbreak.service.cluster.flow

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

import com.sequenceiq.cloudbreak.service.ClusterBasedStatusCheckerTask

@Component
class AmbariDFSSpaceRetrievalTask : ClusterBasedStatusCheckerTask<AmbariClientPollerObject>() {
    var dfsSpace: Map<String, Map<Long, Long>>? = null
        private set

    override fun checkStatus(ambariClientPollerObject: AmbariClientPollerObject): Boolean {
        try {
            dfsSpace = ambariClientPollerObject.ambariClient.dfsSpace
            return true
        } catch (ex: Exception) {
            LOGGER.warn("Error during getting dfs space from ambari", ex)
            return false
        }

    }

    override fun handleTimeout(ambariClientPollerObject: AmbariClientPollerObject) {
    }

    override fun successMessage(ambariClientPollerObject: AmbariClientPollerObject): String {
        return "Dfs space successfully get from ambari."
    }

    companion object {
        val AMBARI_RETRYING_INTERVAL = 5000
        val AMBARI_RETRYING_COUNT = 3

        private val LOGGER = LoggerFactory.getLogger(AmbariDFSSpaceRetrievalTask::class.java)
    }
}
